addons/web/static/src/legacy/utils.js

```diff
@@ -172,12 +172,12 @@ export function makeLegacyRPC(wowlRPC) {
             rpcPromise = wowlRPC(route, args, options);
             rpcPromise
                 .then(function (result) {
-                    if (!target.isDestroyed()) {
+                    if (!target.isDestroyed?.()) {
                         resolve(result);
                     }
                 })
                 .catch(function (reason) {
-                    if (!target.isDestroyed()) {
+                    if (!target.isDestroyed?.()) {
```

Comment:
Hello @sdegueldre , this is a forward port of a fix. I found that the `rpc` function returned for legacy now having four parameters so I changed my fix in `widgt_link.js` to `_rpc: ({ route, params }) => this.rpc(route, params, undefined, this),` and I added this `?.` to cover the case where `isDestroyed` is not defined

Another possibility is to change the condition to `if (target &&target !target.isDestroyed?.())`, which we will keep the rpc calling as `_rpc: this.rpc.bind(this),`

Could you let me know if this is a legit change and do we wanna have the target check too? Thanks! :pray: 
cc: @mcm-odoo

Reply:
You're getting confused here, the code you modified calls the *new* rpc service.

Reply:
I guess I'm the one getting confused, sorry about that :sweat_smile: 

At this point it seems weird that the rpc service changed from the widget version, to the new owl version and now in master it's back to the legacy owl version.

Regardless, I wonder why you need to account for `isDestroyed` being undefined?

Reply:
Hello @sdegueldre no worries! I was also confused when found there were two different `rpc` :rofl: Indeed the fakeWidget has that method, but the `target` here is actually the LinkTools which doesn't have the `isDestroyed`. I know it's bit confusing so I attached a screenshot :pray: 
![image](https://github.com/odoo/odoo/assets/23055668/e4057bcd-b9ea-44d6-99bd-28c3d98d9392)

Reply:
Thanks for checking! @mcm-odoo , shall I do a fix as you mentioned in this PR? :pray:

Reply:
Hello @Jinjiu96 
Yes, you can add the fix in your PR <https://github.com/odoo-dev/odoo/commit/c792be10f1175d0424aba90bcfe0a4ea55f610c3>

Reply:
Just pushed, thanks a bunch! :grin: